main.tf

```diff
@@ -1,3 +1,23 @@
+data "aws_subnets" "istio_private" {
```

Comment:
it's more reliable to have `istio_private_subnet_ids` and `istio_public_subnet_ids`.